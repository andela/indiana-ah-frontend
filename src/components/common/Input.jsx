import React from 'react';
import PropTypes from 'prop-types';
import { InputGroup, FormControl } from 'react-bootstrap';

const Input = ({ placeholder, type, id, value, handleChange }) => (
  <InputGroup >
    <FormControl
      id={id}
      type={type}
      value={value}
      placeholder={placeholder}
      onChange={handleChange}
      required
    />
  </InputGroup >
);

Input.propTypes = {
  label: PropTypes.string,
  labelText: PropTypes.string,
  type: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,

};
export default Input;