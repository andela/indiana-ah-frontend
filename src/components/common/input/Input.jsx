import styled from 'styled-components';
// Create an Input component that'll render an <input> tag with some styles
const Input = styled.input`
  display:flex;
  padding: 0.8rem;
  margin: ${props => props.margin || '0.8rem auto'};
  height: 4.3rem;
  fill: #FBFCFF;
  border: 0.05rem solid rgba(0, 0, 0, 0.4);
  border-radius: 0.4rem;
  width: ${props => props.inputWidth || '94%'};
  box-sizing: border-box;
  @media (max-width: 76.8rem) {
    flex-direction: column;
    height: 3.5rem;
    font-size: 1.5rem
  }
`;

export default Input;
