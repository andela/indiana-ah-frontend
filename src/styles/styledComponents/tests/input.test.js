import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import InputField from '../../../components/common/Input.jsx';
import Input from '../Input.jsx';

describe('Custom Input component', () => {
  const props = {
    type: 'email',
    id: 'email',
    value: 'Odinks',
    placeholder: 'Enter your email',
    handleChange: () => 2,
  };
  const wrapper = shallow(<InputField {...props}/>);
  it('should create an instance and render correctly', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('div')).toHaveLength(1);
    wrapper.setProps({ ...props });
  });
});

describe('Custom Input styled-component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
