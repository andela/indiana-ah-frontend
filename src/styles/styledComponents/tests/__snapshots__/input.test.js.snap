// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input styled-component renders correctly 1`] = `
<input
  className="sc-bdVaJa hZehcL"
/>
`;
